src/Business/StateMachine/Timeout.php

```diff
@@ -151,4 +164,28 @@ protected function validateTimeout(DateInterval $interval, string $timeout): int
 
         return $intervalSum;
     }
+
+    /**
+     * @param \StateMachine\Dto\StateMachine\ItemDto $itemDto
+     * @param \StateMachine\Business\Process\EventInterface $event
+     *
+     * @return void
+     */
+    protected function eventRepeatAction(ItemDto $itemDto, EventInterface $event): void
+    {
+        $moduloValue = (int)Configure::read('StateMachine.eventRepeatAction');
+        if (!$moduloValue) {
+            return;
+        }
+
+        $eventName = $event->getName();
+        $eventName .= $event->getEventTypeLabel();
+
+        $count = $this->transitionLog->getEventCount($eventName, [$itemDto]);
+        if ($count < $moduloValue || $count % $moduloValue !== 0) {
```

Comment:
this will fire an event in case of no repetitions at all - (0 < X || 0!=0)
```suggestion
        if ($count == 0 || $count % $moduloValue !== 0) {
```